src/indentation.rs

```diff
@@ -45,18 +45,21 @@
 /// ```
 /// use textwrap::indent;
 ///
-/// assert_eq!(indent(" \t  Foo   ", "->"), "-> \t  Foo   \n");
+/// assert_eq!(indent(" \t  Foo   ", "->"), "-> \t  Foo   ");
 /// ```
 pub fn indent(s: &str, prefix: &str) -> String {
-    let mut result = String::new();
-    for line in s.lines() {
-        if line.chars().any(|c| !c.is_whitespace()) {
-            result.push_str(prefix);
-            result.push_str(line);
+    let mut output = String::new();
+
+    for line in s.split_inclusive('\n') {
+        if line.trim_start().len() == 0 {
+            output.push_str(line);
+        } else {
+            output.push_str(prefix);
+            output.push_str(line);
```

Comment:
Perhaps it would be a tiny bit simpler to do

```
if (!line.trim().is_empty()) {
  output.push_str(prefix);
}
output.push_str(line);
```

I think that should do the same: conditionally add the `prefix`, unconditionally add the `line` including any trailing whitespace in the `line`.
```

Reply:
Will fix, brainfart here

Reply:
Well, it's almost the same in both cases :smile: